buf/registry/module/v1beta1/commit_service.proto

```diff
@@ -77,11 +77,12 @@ message ResolveCommitsRequest {
   // Once the resource is resolved, this reference is interpeted to refer to the following Commit:
   //   - If a Module is referenced, this is interpreted to mean the latest released Commit on the Module.
   //   - If a Commit is referenced, this just references that specific commit.
-  //   - If a Tag is referenced, this is interpreted to mean the Commit associated with the Tag.
+  //   - If a Tag is referenced, this is interpreted to mean the released Commit associated with the Tag.
```

Comment:
Why released?

Reply:
Same reason as digests, we don't want to allow pinning to something that isn't released.

I wonder if this should be a CLI concern, though (for this case and for the Digest case below).

Reply:
That's a caller implementation detail unless I'm missing something. Should not affect the API.

Reply:
Sure, so just to be clear, I would update both comments, for Tags and for Digests, right?

Reply:
What we choose to allow pinning to in our business logic/implementetion has no implications on this API endpoint. The API does not take any positions on what it chosen to be implemented on top of it. If your specific implementation needs to rely on a Commit being released, it can check the release status of a Commit.

If I reference a Tag, I get back the Commit for that Tag, released or not. We have defined Digest (if I remember) to only mean released commits, although perhaps we should revisit if that restriction is necessary.

Reply:
Agreed that whether a commit is released can be a caller concern. Aside from how tricky this is going to get in some cases (i.e., how do I ask for the latest released commit on a branch), I am trying to be consistent and am finding these rules inconsistent:

> If a Module is referenced, this is interpreted to mean the latest released Commit on the Module.

Okay, so this is released commits only.

> If a Commit is referenced, this just references that specific commit.

Okay, I guess this is fair, but this begs the question why you're trying to resolve a Commit. I find this part of the API very strange.

> If a VCSCommit is referenced, this is interpreted to mean the Commit associated with the VCSCommit.

Okay, so this is not released commit.

> Is a Branch is referenced, this is interpreted to mean the latest Commit on the Branch.

Assuming we go ahead with the new buf sync mental model, is this going to be constrained to a released commit?

> If a Digest is referenced, this is interpreted to mean the latest released Commit that has this Digest.

Okay, so this is released commits only.

Why are we sometimes constraining to released commits and sometimes not?